### PR TITLE
media: modify error types for recorder

### PR DESCRIPTION
**Pull request by jsdosa**
Integrate the error type of the recorder

Signed-off-by: jaesick.shin <jaesick.shin@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [208af515d8d2e0124ce449dcb78d2643d551ef34] - Code Rule Check Result:
framework/include/media/MediaRecorder.h:174: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:174: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:174: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:174: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:174: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:174: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorder.h:185: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:185: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:185: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:185: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorder.h:193: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/media/MediaRecorder.h:193: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/media/MediaRecorder.h:200: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:200: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:200: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)

**Comment by seinfra** [comment]
Target : [208af515d8d2e0124ce449dcb78d2643d551ef34] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [805c1f5e6edadc1c2b8f0125197588d7acdd05ba] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [805c1f5e6edadc1c2b8f0125197588d7acdd05ba] - Code Rule Check Result:
framework/include/media/MediaRecorder.h:175: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:175: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:175: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:175: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:175: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:175: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorder.h:186: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:186: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:186: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:186: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorder.h:194: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/media/MediaRecorder.h:194: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/media/MediaRecorder.h:201: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:201: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:201: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)

File: framework/include/media/MediaRecorder.h [hunk 1]
```diff
@@ -43,27 +43,21 @@ class MediaRecorderImpl;
  * @details @b #include <media/MediaRecorder.h>
  * @since TizenRT v2.0 PRE
  */
-typedef enum recorder_error_e : int {
+enum recorder_error_e : int {
 	/** MediaRecorder Error case */
-	RECORDER_ERROR_NOT_ALIVE = -16,
+	RECORDER_ERROR_NOT_ALIVE = INT8_MIN,
 	RECORDER_ERROR_INVALID_STATE,
```
Comment (juitem) [inline framework/include/media/MediaRecorder.h]: why int8_MIN?

File: framework/include/media/MediaRecorderObserverInterface.h [hunk 2]
```diff
@@ -29,11 +29,14 @@
 #ifndef __MEDIA_MEDIARECOREROBSERVERINTERFACE_H
 #define __MEDIA_MEDIARECOREROBSERVERINTERFACE_H
 
-#include <media/MediaRecorder.h>
+//#include <media/MediaRecorder.h>
```
Comment (chanijjani) [inline framework/include/media/MediaRecorderObserverInterface.h]: How about deleting this or marking ToDo?

File: framework/include/media/MediaRecorder.h [hunk 3]
```diff
@@ -43,27 +43,21 @@ class MediaRecorderImpl;
...
-	RECORDER_ERROR_SET_VOLUME_AUDIO_FAILED,
-	RECORDER_ERROR_GET_VOLUME_AUDIO_FAILED,
 	/** MediaRecorder Success case */
-	RECORDER_ERROR_NONE = 0
-} recorder_error_t;
+	RECORDER_ERROR_NONE = 0,
+	RECORDER_OK = 0
+};
+
+typedef enum recorder_error_e recorder_result_t;
```
Comment (junmin-kim) [inline framework/include/media/MediaRecorder.h]: how about
```
    RECORDER_ERROR_NONE = 0,
};
typedef enum recoreder_error_e recoreder_error_t;
const int RECORDER_OK = RECORDER_ERROR_NONE;
typedef int recorder_result_t;
```

File: framework/include/media/MediaRecorderObserverInterface.h (hunk 2 above)
Comment (junmin-kim) [inline framework/include/media/MediaRecorderObserverInterface.h]: we can remove this line

**Comment by seinfra** [comment]
Target : [371861faec8c82d57d2095e48518eaa821737eab] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [371861faec8c82d57d2095e48518eaa821737eab] - Code Rule Check Result:
framework/include/media/MediaRecorder.h:178: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:178: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:178: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:178: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:178: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:178: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorder.h:189: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:189: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:189: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:189: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorder.h:197: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/media/MediaRecorder.h:197: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/media/MediaRecorder.h:204: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:204: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:204: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:65: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:71: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:77: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:83: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:89: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:95: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)

File: framework/include/media/MediaRecorder.h (hunk 3 above)
Comment (jsdosa) [reply framework/include/media/MediaRecorder.h]: it is good. i will change that.

File: framework/include/media/MediaRecorderObserverInterface.h (hunk 2 above)
Comment (jsdosa) [reply framework/include/media/MediaRecorderObserverInterface.h]: i will fix it.

**Comment by seinfra** [comment]
Target : [efaa43531448cbe9b3a3a2435eb16ee55fc0e1c5] - Code Rule Check Result:
framework/include/media/MediaRecorder.h:179: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:179: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:179: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:179: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:179: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:179: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorder.h:190: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaRecorder.h:190: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:190: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:190: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorder.h:198: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/include/media/MediaRecorder.h:198: ERROR: [IDT_M_TAB] please, no space before tabs
framework/include/media/MediaRecorder.h:205: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaRecorder.h:205: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/MediaRecorder.h:205: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:64: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:70: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:76: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:82: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:88: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaRecorderObserverInterface.h:94: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:66: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/MediaRecorderImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:66: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxE)
framework/src/media/MediaRecorderImpl.h:69: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:83: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:85: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:85: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:85: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:85: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:92: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:93: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:94: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:95: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:99: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:100: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:101: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:101: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:101: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:101: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:102: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:102: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:102: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:102: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:102: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:102: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:102: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:103: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:106: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:106: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:106: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:107: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:107: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:107: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/MediaRecorderImpl.h:107: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:107: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaRecorderImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaRecorderImpl.h:108: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:110: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaRecorderImpl.h:114: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

**Comment by seinfra** [comment]
Target : [efaa43531448cbe9b3a3a2435eb16ee55fc0e1c5] - Code Rule Check (C++) OK.

File: framework/src/media/MediaRecorderImpl.cpp [hunk 4]
```diff
@@ -367,7 +367,7 @@ recorder_error_t MediaRecorderImpl::getVolume(uint8_t *vol)
 	return ret;
 }
 
-void MediaRecorderImpl::getRecorderVolume(uint8_t *vol, recorder_error_t &ret)
+void MediaRecorderImpl::getRecorderVolume(uint8_t *vol, recorder_result_t &ret)
```
Comment (JeongJunSik) [inline framework/src/media/MediaRecorderImpl.cpp]: recorder_result_t& ret

**Pull request merged**